srv/modules/runners/replace.py

```diff
@@ -0,0 +1,161 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=too-few-public-methods,modernize-parse-error
+"""
+Runner to remove a single osd
+"""
+
+from __future__ import absolute_import
+from __future__ import print_function
+import time
+import logging
+import os
+# pylint: disable=import-error,3rd-party-module-not-gated,redefined-builtin
+import salt.client
+import salt.runner
+
+log = logging.getLogger(__name__)
+
+
+def help_():
+    """
+    Usage
+    """
+    usage = ('salt-run replace.osd id [id ...][force=True]:\n\n'
+             '    Removes an OSD from a minion\n'
+             '\n\n')
+    print(usage)
+    return ""
+
+
+def osd(*args, **kwargs):
+    """
+    Remove an OSD gracefully or forcefully on the minion
+
+    Note: If I were not in Salt, this whole routine would become a library.
+    This runner can be called by remove.osd and is only different by three
+    commands.  However, we have one runner call and multiple module calls below.
+    Trying to refactor these module calls into a __utils__ would require yet
+    another module.  Debugging all of this at runtime would be add to the fun,
+    so I didn't.
+    """
+    # Parameters for osd.remove module
+    supported = ['force', 'timeout', 'delay']
+    passed = ["{}={}".format(k, v) for k, v in kwargs.items() if k in supported]
+    log.debug("Converted kwargs: {}".format(passed))
+
+    # OSDs to remove
+    osds = list(str(arg) for arg in args)
+    if _checks_failed(osds, kwargs):
+        return ""
+
+    master_minion = _master_minion()
+
+    local = salt.client.LocalClient()
+    host_osds = local.cmd('I@roles:storage', 'osd.list', tgt_type='compound')
+
+    completed = osds
+    for osd_id in osds:
+        host = _find_host(osd_id, host_osds)
+        if host:
```

Comment:
We should be verbose in case there was no host found. Currently it will just not do anything and bail out silently

Reply:
The remove.osd command also calls this module.  In the case that a remove.osd has already been called and is no longer present on the minions, then the OSD will be removed from Ceph (a message is printed for that).  Currently, the behavior prints messages for what actions it does.  

I could add a verbose option, but I would prefer the default to not be verbose.  If an admin is using a wildcard for dozens of OSDs and rerunning the command multiple times, then the output gets shorter as the Ceph cluster reaches the desired state.